### PR TITLE
Various minor cleanup

### PR DESCRIPTION
**Pull request by g2flyer**
**What this PR does / why we need it**:
This PR cleans-up a few things, mostly related to our move to go modules and to fabric v2
- remove go gets from Dockerfile which got obsoleted by go mod
- run 'go mod tidy' after build
- fixed remaining references to old go shim
- added pointer to auction demo description/spec also to top-level demo README.md

### CODE REVIEW

File: demo/client/backend/mock/chaincode/fpc_chaincode.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ package chaincode
 
 import (
 	"github.com/hyperledger-labs/fabric-private-chaincode/ecc"
-	"github.com/hyperledger/fabric/core/chaincode/shim"
+        "github.com/hyperledger/fabric-chaincode-go/shim"
```
Comment (mbrandenburger) [inline demo/client/backend/mock/chaincode/fpc_chaincode.go]: is this correctly indented?

File: examples/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
+Refer to [package shim](github.com/hyperledger/fabric-chaincode-go/shim) for the GoDoc for shim interface provided by Fabric.  The FPC programming model for chaincode provides a simpler version of the _shim_ SDK provided by Hyperledger Fabric in Go and node.js.  FPC provides a C++ interface  to access its state variables and transaction context through [shim.h](https://github.com/hyperledger-labs/fabric-private-chaincode/blob/master/ecc_enclave/enclave/shim.h).  The standard commands are similar to the ones in Fabric.  To ensure confidentiality of the arguments passed to the chaincode, the arguments are encrypted while using FPC SDK.
```
Comment (mbrandenburger) [inline examples/README.md]: good catch!

File: demo/client/backend/mock/chaincode/fpc_chaincode.go (hunk 1 above)
Comment (mbrandenburger) [reply demo/client/backend/mock/chaincode/fpc_chaincode.go]: this fixes a left over from #247

Comment (MHBauer) [reply demo/client/backend/mock/chaincode/fpc_chaincode.go]: got tabs and spaces mixed here. probably needs a gofmt.

Comment (g2flyer) [reply demo/client/backend/mock/chaincode/fpc_chaincode.go]: Indeed. Given that demo is now disabled in the build, make checks didn't catch it.  Just pushed an update

**Pull request merged**